Added Feature Cover Json Data Node in Data Node control

THEBOSS0369 opened the pull request:
This PR fixes #1248 

Hey Everyone!

In this PR, The changes has been done in `DataNodeViewer.tsx` file, Now Data in Json can be collapsed or expand as the user wish. 
I have imported `react-json-view` because it will make our process much easier.

Thanks!

THEBOSS0369 committed:
Added Feature Cover Json Data Node in Data Node control

THEBOSS0369 commented:
@quest-bot loot #1248

FredLL-Avaiga commented on frontend/taipy/src/DataNodeViewer.tsx:
```diff
@@ -56,6 +56,7 @@ import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
 import { AdapterDateFns } from "@mui/x-date-pickers/AdapterDateFnsV3";
 import { format } from "date-fns";
 import deepEqual from "fast-deep-equal/es6";
+import ReactJson from 'react-json-view';
```
Should be lazy loaded

FredLL-Avaiga commented on frontend/taipy/src/DataNodeViewer.tsx:
```diff
@@ -756,6 +757,29 @@ const DataNodeViewer = (props: DataNodeViewerProps) => {
                                 ) : null}
                             </Stack>
                         </Box>
+                        <div>
```
This should go in the data tab

FredLL-Avaiga commented on frontend/taipy/src/DataNodeViewer.tsx:
```diff
@@ -756,6 +757,29 @@ const DataNodeViewer = (props: DataNodeViewerProps) => {
                             </Stack>
                         </Box>
+                        <div>
+                            { dtType === "json" ? (
```
Did you check that this type is set by the backend ?

THEBOSS0369 replied:
No i didn't checked as on the issue the user was using json data so i thought i will make it in a condition so that only json data will be collapsed.

FredLL-Avaiga replied:
Here is what is a [taipy.core.data_node ](https://docs.taipy.io/en/latest/userman/scenario_features/data-integration/data-node-usage/)

A data_node can contain any type of data (scalar, tabular, structured, db view ...)
the data_node component tries to display a data_node and its underlying data.

FredLL-Avaiga replied:
as per https://github.com/Avaiga/taipy/blob/63f0cbf9a5e6346e54c5c4c0071ee8dc2cca8828/taipy/gui_core/_adapters.py#L205

the best we can hope for is a `dict` value for `dtType` and that's what we get if the `DataNode` used is a `JSONDataNode`.
Described here
https://github.com/Avaiga/taipy/blob/9373005365e32d9bbc70d05f455c32475c7cf44e/taipy/core/data/json.py#L141

THEBOSS0369 replied:
Thanks for sharing this i was having a really hard time with the backend.

FredLL-Avaiga reviewed (changes requested):
Needs tests

THEBOSS0369 committed:
fixed the requested changes

THEBOSS0369 commented:
> Needs tests

Hey @FredLL-Avaiga ! Apologies for late reply. I have created a test file can you tell me in which folder should i add that file , this is the test file in the root folder

![image](https://github.com/user-attachments/assets/722a3ce6-414e-4e8d-9efb-bae76639b829)

FredLL-Avaiga commented:
> > Needs tests
> 
> Hey @FredLL-Avaiga ! Apologies for late reply. I have created a test file can you tell me in which folder should i add that file , this is the test file in the root folder
> 
> ![image](https://private-user-images.githubusercontent.com/145831712/381195143-722a3ce6-414e-4e8d-9efb-bae76639b829.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyMTc3NzMsIm5iZiI6MTczMDIxNzQ3MywicGF0aCI6Ii8xNDU4MzE3MTIvMzgxMTk1MTQzLTcyMmEzY2U2LTQxNGUtNGU4ZC05ZWZiLWJhZTc2NjM5YjgyOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyOVQxNTU3NTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOTgwYzMyMzdhZjVmZjZjNTc2Yjg1MDlhNTMwNmQ4MTUyMDYzYWU1OTEwMzQwZDQ1ZjhhYTc2ODFhNmJhZDEyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.MUgp6qf8eHFtoYH6nQVA1ul_D3Ri3udHvuX5PGyG_lc)

What does this show ? that you can display some json data in a markdown page ?
I don't understand how this would use the data_node control ?

FredLL-Avaiga reviewed (changes requested):
this needs:
- ~~backend check and modification if necessary~~
- tests
- example of the data_node control usage with json

FredLL-Avaiga commented:
please fix the failing checks

THEBOSS0369 commented:
> please fix the failing checks

Yes sir i will fix these errors.

THEBOSS0369 committed:
Adding example file

THEBOSS0369 commented:
@FredLL-Avaiga The test is failing because of the `<Grid >` component as in file it has been imported from `@mui/material/Grid2`
and when i'm using it this error is showing 
![image](https://github.com/user-attachments/assets/895a0094-57c9-4ccd-b828-33eeb63531c4)

it can resolve if i import it from `@mui/material/Grid` which is definitely not good option , so can you tell me how to use Grid in this file.

THEBOSS0369 commented:
> * example of the data_node control usage with json

Sir, I will be adding this in the next commit

FredLL-Avaiga commented:
> @FredLL-Avaiga The test is failing because of the `<Grid >` component as in file it has been imported from `@mui/material/Grid2`
> and when i'm using it this error is showing 
> ![image](https://github.com/user-attachments/assets/895a0094-57c9-4ccd-b828-33eeb63531c4)
> 
> it can resolve if i import it from `@mui/material/Grid` which is definitely not good option , so can you tell me how to use Grid in this file.

You have to update your dependencies
Grid2 comes with Mui6 which is what we use in Taipy

THEBOSS0369 commented:
> > @FredLL-Avaiga The test is failing because of the `<Grid >` component as in file it has been imported from `@mui/material/Grid2`
> > and when i'm using it this error is showing 
> > ![image](https://github.com/user-attachments/assets/895a0094-57c9-4ccd-b828-33eeb63531c4)
> > 
> > it can resolve if i import it from `@mui/material/Grid` which is definitely not good option , so can you tell me how to use Grid in this file.
> 
> You have to update your dependencies
> Grid2 comes with Mui6 which is what we use in Taipy 

Got it Sir , This will work! Also I won't be able to share any update till tomorrow due to Diwali Festival. Hope you understand 😀

FredLL-Avaiga commented:
> Also I won't be able to share any update till tomorrow due to Diwali Festival. Hope you understand 😀

Enjoy the festival 🙏

FredLL-Avaiga commented on doc/gui/examples/controls/collapse_data_view.py:
```diff
@@ -0,0 +1,25 @@
+
+# Define the page with the JSON viewer
+page = """
+<|{json_data}|json|expandable|>
```
did you try to run this ?
This should be something like

```
import taipy as tp
import taipy.gui.builder as tgb
from taipy import Config, Orchestrator, Scope
from taipy.gui import Gui

json_config_node = Config.configure_json_data_node(
    id="json_node",
    default_path="./demo-1248-dn.json",
    scope=Scope.GLOBAL,
)

with tgb.Page() as data_node_viewer:
    tgb.html("h2", "Discovering Data Node json viewer")

    tgb.data_node(
        data_node="{json_data_node}"
    )
gui = Gui(page=data_node_viewer)

if __name__ == "__main__":
    Orchestrator().run()
    json_data_node = tp.create_global_data_node(json_config_node)
    gui.run(title="1248 json data_node")
```

THEBOSS0369 replied:
Yes sir , I did try running that. This is what it was showing (we didn't wanted this)
![image](https://github.com/user-attachments/assets/d8120353-40a4-4284-a5ae-e9b590e5bf44)

FredLL-Avaiga reviewed (changes requested):
to make things work, you will nee to modify the backend also:
in _adapters.py
https://github.com/Avaiga/taipy/blob/63f0cbf9a5e6346e54c5c4c0071ee8dc2cca8828/taipy/gui_core/_adapters.py#L190

the static method should be
```py
    @staticmethod
    def _is_tabular_data(datanode: DataNode, value: t.Any):
        return isinstance(datanode, _TabularDataNodeMixin) or (
            isinstance(value, (pd.DataFrame, pd.Series, list, tuple, dict)) and not isinstance(datanode, JSONDataNode)
        )
```

PS do not forget to import `JSONDataNode`
`from taipy.core.data import JSONDataNode`

FredLL-Avaiga commented on frontend/taipy/src/DataNodeViewer.tsx:
```diff
@@ -1026,6 +1030,28 @@ const DataNodeViewer = (props: DataNodeViewerProps) => {
+                                                    <>
+                                                    <Suspense fallback={<div>Loading JSON...</div>}>
+                                                        <Grid container spacing={2}>
+                                                            <Grid xs={12}>
```
change `xs={12}` to `size={12}`

THEBOSS0369 replied:
Done

FredLL-Avaiga reviewed (changes requested):
You should also have a change in package.json ?
and the package `react-json-view` is not maintained any more ... need to change it

THEBOSS0369 committed:
Fixed the requested Changes

THEBOSS0369 commented:
@FredLL-Avaiga ! As `react-json-view` is not maintained now so i have imported `react-json-tree ` which is maintained regularly
and takes very much less space comparted to `react-json-view` i have created 2 files to import that because when i was lazy importing it i was getting many errors , to avoid that i divided in file `LazyReactJsonTree.tsx` and `react-json-tree.ds`. All the errors that i was getting get resolved by this

I created a demo-1248-dn.json file for the demo i have added the same file in root folder for testing as well, but don't know why its not showing when i try to add it in the commit here is the ss for it, I think i am adding that file in wrong folder maybe that is causing problem? 
![image](https://github.com/user-attachments/assets/c489e5fe-965d-4663-a487-36c75577cca2)

 the output is coming something like this
![image](https://github.com/user-attachments/assets/477e2277-27fb-4f9a-9290-6572ffcd253c)

THEBOSS0369 committed:
fixing linter error

FredLL-Avaiga commented:
We don't want any demo-* fille in the repo
But at least you can run it and test the result of your changes sand debug the frontend if necessary.

FredLL-Avaiga commented:
By the screen shot it doesn't look like the json is rendered?

THEBOSS0369 commented:
> By the screen shot it doesn't look like the json is rendered?

Yes sir I don't know what am i doing wrong there is a popup comes when i hover over Data that shows `blocked by me` . i don't understand  why its coming
![image](https://github.com/user-attachments/assets/49daa27e-348d-4ed2-845a-4b48ab82aa6f)

i also tried it by changing `shouldExpandNode={() => false}` to `true ` still nothing changes. Can you help me with what am i doing wrong

FredLL-Avaiga commented:
You'll have to debug the code 
Hint: the json viewer is not where it should be in the tree..
It should be close to the display value when in non editable mode

somewhere near 
https://github.com/Avaiga/taipy/blob/aea1cee3109ad4b98674e00a51d17468114e910f/frontend/taipy/src/DataNodeViewer.tsx#L1111

THEBOSS0369 committed:
added the code in display mode

THEBOSS0369 commented:
@FredLL-Avaiga Can you look at it please that this time i'm on the right track or not , Although the json DATA is still not rendering, if i have done right this time then i will do some tries maybe that work, If i am not doing right please let me know .

FredLL-Avaiga commented:
> @FredLL-Avaiga Can you look at it please that this time i'm on the right track or not , Although the json DATA is still not rendering, if i have done right this time then i will do some tries maybe that work, If i am not doing right please let me know .

Are you testing your code?

FredLL-Avaiga commented on frontend/taipy/package.json:
```diff
@@ -30,7 +30,9 @@
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "react-json-tree": "^0.19.0",
+    "taipy-gui": "file:../../taipy/gui/webapp"
```
This line should never be committed

THEBOSS0369 replied:
> This line should never be committed

I will remove this line it probably occurred when I installed `react-json-views`

FredLL-Avaiga reviewed (changes requested):
The code modification should impact only DatanodeViewer.tsx

THEBOSS0369 commented:
> > @FredLL-Avaiga Can you look at it please that this time i'm on the right track or not , Although the json DATA is still not rendering, if i have done right this time then i will do some tries maybe that work, If i am not doing right please let me know .
> 
> Are you testing your code?

Yes sir i tested the code after doing the changes it still isn't working so before moving on and trying out different options I was just confirming that this time the code that i added is in the right place or not.

FredLL-Avaiga commented:
> > @FredLL-Avaiga The test is failing because of the `<Grid >` component as in file it has been imported from `@mui/material/Grid2`
> > and when i'm using it this error is showing
> > ![image](https://private-user-images.githubusercontent.com/145831712/381860430-895a0094-57c9-4ccd-b828-33eeb63531c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA1NDg4ODIsIm5iZiI6MTczMDU0ODU4MiwicGF0aCI6Ii8xNDU4MzE3MTIvMzgxODYwNDMwLTg5NWEwMDk0LTU3YzktNGNjZC1iODI4LTMzZWViNjM1MzFjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwMlQxMTU2MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NWExYjk5ODcyMGE2NTc3YjUzYmE3N2Y1ZGE0M2QwODgxNjVjMTMxOTQ4NDUyNjdhZWFhY2I0MjMxYWEzZGU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._6KkBKY-4WYRhM6Z2md4BFQr8f2-LjBaHmo6S8m7xBQ)
> > it can resolve if i import it from `@mui/material/Grid` which is definitely not good option , so can you tell me how to use Grid in this file.
> 
> You have to update your dependencies Grid2 comes with Mui6 which is what we use in Taipy

You do use Grid2 but Grid2 doesn't use xs by size now ...

FredLL-Avaiga commented on frontend/taipy/src/react-json-tree.d.ts:
I would prefer a package that provides its own types and that is maintained and compatible with the react version we use

FredLL-Avaiga commented on frontend/taipy/src/utils/LazyReactJsonTree.tsx:
please change package

FredLL-Avaiga commented on frontend/taipy/package.json:
```diff
@@ -30,7 +30,9 @@
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "react-json-tree": "^0.19.0",
```
please change package

FredLL-Avaiga commented on frontend/taipy/src/DataNodeViewer.tsx:
```diff
@@ -1110,6 +1113,14 @@ const DataNodeViewer = (props: DataNodeViewerProps) => {
                                                             disabled={true}
                                                             title={`${dtValue}`}
                                                         />
+                                                    ) : dtType === "json" ? (
```
`json` should be `dict` (check _adapters.py

FredLL-Avaiga reviewed (changes requested):
get a maintained json viewer/editor

THEBOSS0369 commented:
> get a maintained json viewer/editor

Gotcha Sir! I will do the changes and come back to you

THEBOSS0369 commented:
@FredLL-Avaiga ! I found this one called json-viewer by textealnc https://github.com/TexteaInc/json-viewer this is being maintained properly plus 100 percent typescript and we can use it with npm package so i'm going with it. If you think its not suitable do let me know.

THEBOSS0369 committed:
added new library

THEBOSS0369 commented:
@FredLL-Avaiga please check now the files i have added. If this package still doesn't fulfill the requirement then i have found another one this is the lite version of the first one i added, this is called `react-json-view-lite` https://www.npmjs.com/package/react-json-view-lite

FredLL-Avaiga commented on frontend/taipy/package.json:
```diff
@@ -34,6 +34,7 @@
     "@mui/x-date-pickers": "^7.0.0",
     "@mui/x-tree-view": "^7.0.0",
     "@projectstorm/react-diagrams": "^7.0.2",
+    "@textea/json-viewer": "^4.0.0",
```
Based in Mui. Good choice 👍

FredLL-Avaiga commented on frontend/taipy/src/react-json-viewer.d.ts:
No need for a separate file

FredLL-Avaiga commented on frontend/taipy/src/utils/LazyJsonViewer.tsx:
Put all this in the datanode viewer

FredLL-Avaiga reviewed (changes requested):
Keep everything in the datanode viewer

THEBOSS0369 committed:
Moved importation of library in single file

THEBOSS0369 commented:
Hey @FredLL-Avaiga ! Please check now i have moved everything in the single file.

FredLL-Avaiga commented on frontend/taipy/package.json:
```diff
@@ -34,11 +34,13 @@
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "taipy-gui": "file:../../taipy/gui/webapp"
```
please do not commit this line

FredLL-Avaiga commented on frontend/taipy/src/DataNodeViewer.tsx:
```diff
@@ -95,7 +97,9 @@ import { useUniqueId } from "./utils/hooks";
-const editTimestampFormat = "yyyy/MM/dd HH:mm";
+const JsonViewer = lazy(() => import("@textea/json-viewer").then(module => ({ default: module.JsonViewer })));
+
+const editTimestampFormat = "YYY/MM/dd HH:mm";
```
this was changed in the develop branch
https://github.com/Avaiga/taipy/blob/aea1cee3109ad4b98674e00a51d17468114e910f/frontend/taipy/src/DataNodeViewer.tsx#L98

FredLL-Avaiga commented on frontend/taipy/src/DataNodeViewer.tsx:
```diff
@@ -1119,6 +1123,17 @@ const DataNodeViewer = (props: DataNodeViewerProps) => {
                                                             disabled={true}
                                                             title={`${dtValue}`}
                                                         />
+                                                    ) : dtType === "dict" ? (
```
I think this is the right spot

FredLL-Avaiga commented on frontend/taipy/src/DataNodeViewer.tsx:
```diff
@@ -1119,6 +1123,17 @@ const DataNodeViewer = (props: DataNodeViewerProps) => {
+                                                            <Grid size={12}>
+                                                                <Typography variant="subtitle2">JSON Data</Typography>
+                                                            </Grid>
+                                                            <Grid size={12}>
```
we don't need those lines

FredLL-Avaiga commented on frontend/taipy/src/DataNodeViewer.tsx:
```diff
@@ -1119,6 +1123,17 @@ const DataNodeViewer = (props: DataNodeViewerProps) => {
+                                                                <Suspense fallback={<div>Loading JSON Viewer...</div>}>
+                                                                    <JsonViewer value={dtValue} collapsed={1} groupArraysAfterLength={5} />
+                                                                </Suspense>
+                                                            </Grid>
+                                                        </Grid> 
```
delete these lines

THEBOSS0369 committed:
removing uncesseary lines

THEBOSS0369 committed:
Update package.json

THEBOSS0369 committed:
Update DataNodeViewer.tsx

THEBOSS0369 committed:
Update package.json

THEBOSS0369 committed:
Merge branch 'develop' into feature/#1248-cover-json-data-node-control-new

THEBOSS0369 commented:
@FredLL-Avaiga Every requested changes Done.

THEBOSS0369 committed:
Update DataNodeViewer.tsx

THEBOSS0369 commented:
Hey @FredLL-Avaiga ! I have did the requested changes. Once you get the time do let me know what to do next. Thanks !

FredLL-Avaiga commented:
> Hey @FredLL-Avaiga ! I have did the requested changes. Once you get the time do let me know what to do next. Thanks !

you need to fix every failing tests (related to your changes)

THEBOSS0369 committed:
fixing tests

THEBOSS0369 commented:
Hey @FredLL-Avaiga ! It seems the tests are failing because new libraries are missing from my package-lock.json file. So i have added the file let's see if this fix the issue
![image](https://github.com/user-attachments/assets/f3fc390d-8faa-42af-b362-0d449c50004e)

THEBOSS0369 committed:
fixing failing tests again

THEBOSS0369 commented:
@FredLL-Avaiga Sir ! Now the test is failing because of these libraries not found, i checked it in the main codespace i wasn't able to find these 2 lib and they are not related to my changes
![Screenshot 2024-11-07 202455](https://github.com/user-attachments/assets/78705d72-26de-4769-a10f-56e26759d571)

FredLL-Avaiga commented on doc/gui/examples/controls/collapse_data_view.py:
```diff
@@ -0,0 +1,23 @@
+)
+
+with tgb.Page() as data_node_viewer:
+    tgb.html("h2", "Discovering Data Node JSON Viewer")
```
not needed

FredLL-Avaiga commented on doc/gui/examples/controls/collapse_data_view.py:
```diff
@@ -0,0 +1,23 @@
+if __name__ == "__main__":
+    Orchestrator().run()
+    json_data_node = tp.create_global_data_node(json_config_node)
+    gui.run(title="1248 JSON Data Node")
```
title = "Datanode Viewer - json support"

FredLL-Avaiga commented on doc/gui/examples/controls/collapse_data_view.py:
rename datanode_viewer_json.py

FredLL-Avaiga reviewed (changes requested):
For your package issue, you would recommend that you delete your local `package-lock.json` and `node_modules` folder and run `npm i` to regenerate them.

FredLL-Avaiga commented:
I think you will see that :
- JSonViewer does not take a `collapse` property
- it does that a `theme` property that you can set to `theme.palette.mode` after getting the current theme by `const theme = useTheme();`
- you need to block the edition mode for now with 
```
                              <Grid
                                        container
                                        size={12}
                                        justifyContent="space-between"
                                        data-focus={dataValueFocus}
                                        onClick={dtType == "dict" ? undefined: onFocus}
                                        sx={hoverSx}
                                    >
```

around https://github.com/Avaiga/taipy/blob/7dac175ba98ddc80a01f7d63f0390bc74a765f1c/frontend/taipy/src/DataNodeViewer.tsx#L997

ie replacing the onClick handler with undefined when dtType == "dict"

THEBOSS0369 committed:
Added theme and done some requested changes

THEBOSS0369 commented:
Hey @FredLL-Avaiga ! Can you take a look now.

FredLL-Avaiga commented:
> Hey @FredLL-Avaiga ! Can you take a look now.

Can you fix the checks ?
`Property 'collapsed' does not exist on type 'IntrinsicAttributes & JsonViewerProps<unknown>'.`

THEBOSS0369 commented:
@FredLL-Avaiga ! I think this problem is occuring because maybe there is no collapsed function showing in the main lib . So shall i go ahead and use any other lib like this one https://www.npmjs.com/package/react-json-view-lite
![image](https://github.com/user-attachments/assets/fa546335-19c9-41af-b065-1e6ca9814d4e)

FredLL-Avaiga commented:
No just remove the collapse properly

THEBOSS0369 commented:
@FredLL-Avaiga ! It seems all tests have passed what to do next?

FredLL-Avaiga reviewed (dismissed):
Let's see if @FabienLelaquais liked it

FredLL-Avaiga commented:
Can you merge this @jrobinAV ?

THEBOSS0369 commented:
Hey @FredLL-Avaiga ! There was an extra comma in json file which was not letting the test passed now i have removed it and test are Passing.

FredLL-Avaiga commented on doc/gui/examples/controls/datanode_viewer_json.py:
```diff
@@ -0,0 +1,22 @@
+
+json_config_node = Config.configure_json_data_node(
+    id="json_node",
+    default_path="./demo-1248-dn.json",
```
where is the json file ?

FredLL-Avaiga replied:
I guess it should be renamed datanode_viewer_json.json
And that would be why the demo-* file was ignored

THEBOSS0369 committed:
Adding json file

THEBOSS0369 committed:
json file

THEBOSS0369 committed:
Update DataNodeViewer.tsx

THEBOSS0369 commented:
@FredLL-Avaiga  Sir all tests passed if its ready to merge please merge this. Otherwise merge conflict will create the issue 🙂

FredLL-Avaiga reviewed (approved):
Another try @jrobinAV

FredLL-Avaiga committed:
Merge branch 'develop' into feature/#1248-cover-json-data-node-control-new

THEBOSS0369 commented:
@jrobinAV  Thanks for merging this  and @FredLL-Avaiga  thanks for your guidance 😃.

THEBOSS0369 commented:
@RymMichaut am i still eligible for the swags for hacktoberfest 300 points? 😅

RymMichaut commented:
yes of course @THEBOSS0369 you'll receive it with the other package :)
